sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcConnectionString.java

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.jdbc;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.NONE_VALUE;
+
+
+public class JdbcConnectionString {
+
+    static final String INVALID_CONNECTION_STRING_FORMAT = "Invalid connection string: %s";
+    static final String INVALID_PROPERTY_PAIR_FORMAT = "Connection string has invalid key value pair: %s";
+    private static final Logger LOGGER = LoggerFactory.getLogger(JdbcConnectionString.class);
+    private static final String TOKEN_VALUE_SEPARATOR = "=";
+    private final String str;
+    private final Map<String, String> properties = new HashMap<>();
+    private DatabaseType databaseType = null;
+
+    JdbcConnectionString(String str) {
+        this.str = str;
+        resolveSegments();
+    }
+
+    private void resolveSegments() {
+        if (!StringUtils.hasText(this.str)) {
+            LOGGER.warn("'connectionString' doesn't have text.");
+            throw new IllegalArgumentException(String.format(INVALID_CONNECTION_STRING_FORMAT, this.str));
+        }
+
+        Optional<DatabaseType> optionalDatabaseType = Arrays.stream(DatabaseType.values())
+            .filter(databaseType -> this.str.startsWith(databaseType.getSchema()))
+            .findAny();
+
+        if (optionalDatabaseType.isPresent()) {
+            this.databaseType = optionalDatabaseType.get();
+        } else {
+            throw new IllegalArgumentException(String.format(INVALID_CONNECTION_STRING_FORMAT, this.str));
```

Comment:
I'm not sure if we should call this **invalid**, for example, user provides a DB2 JDBC url, it is correct and valid, just this code doesn't **support** DB2

Reply:
yes, will reconsider it.